### ISSUE TITLE
Victorialogs set stream name for syslog input

### ISSUE THREAD

**Issue by alexeynl** [VictoriaMetrics/VictoriaMetrics#7354]
### Is your feature request related to a problem? Please describe

As [documentation states](https://docs.victoriametrics.com/victorialogs/data-ingestion/syslog/#), logs received by syslog  are automaticaly splited to different streams according to 3 steam field

> hostname, app_name and proc_id - stream fields for unique identification over every log stream.

As far as i understand this set of fields can't be changed in case of syslog.

Lets consider we are going to configure two ways to receive logs by victoria logs:
1. Vector elasticsearch sink.
2. Syslog sources send log to specific victorialog syslog port (for example 514).

So we want to have only two streams in victoria logs for quick search what we want. For example we need search only by specific victorialog syslog source. But we cant because each syslog source are splitted to different streams by hostname, app_name and proc_id.

### Describe the solution you'd like

We need option for syslog inputs to specify a stream field name and value that will be considered as stream field during log ingestion. For example all logs ingested by victorialogs to 514 syslog port should have stream field such as stream_name=syslog_nginx. 
In other words we need option for tagging of syslog inputs.

**Comment by valyala** [comment]
> For example we need search only by specific victorialog syslog source. But we cant because each syslog source are splitted to different streams by hostname, app_name and proc_id.

Just assign an additional unique field per each source of logs. For example, all the logs from Vector may have `source=vector` field, while all the logs from Syslog sources may have `source=syslog` field. Then you can filter logs from particular source by `source:=vector` or `source:=syslog` [filters](https://docs.victoriametrics.com/victorialogs/logsql/#exact-filter).

VictoriaLogs doesn't provide an ability to add some `field=value` field to all the logs ingested from a particular source, so this field must be assigned at the source side. I think it would great adding support for `extra_field` query arg to [all the supported HTTP-based data ingestion protocols](https://docs.victoriametrics.com/victorialogs/data-ingestion/#http-apis). If this query arg is set, then the specified extra field must be added to all the ingested logs via the given HTTP request.

As for mixing multiple [log streams](https://docs.victoriametrics.com/victorialogs/keyconcepts/#stream-fields) into a single log stream, this is extremely bad idea, since it will result in the following issues:
- Increased storage space usage, because logs from mixed log streams compress not so good comparing to logs from a single log stream.
- Increased CPU usage during background merge of the ingested logs.
- Degraded query performance, since more logs should be scanned during queries.

The log stream is defined as a stream of logs from a **single application instance**.

**Comment by valyala** [comment]
FYI, the ability to add extra fields to the ingested logs has been added in the commit 4478e48eb65f8296fb63288cd34a4847dddb5a65 . See the description for `extra_fields` query arg and `VL-Extra-Fields` HTTP header in [these docs](https://docs.victoriametrics.com/victorialogs/data-ingestion/#http-parameters) for more details.

This commit will be included in the next release (v0.41.0-victorialogs). In the mean time VictoriaLogs can be built from this commit according to [these instructions](https://docs.victoriametrics.com/victorialogs/quickstart/#building-from-source-code).

**Comment by alexeynl** [comment]
> the ability to add extra fields to the ingested logs has been added

Hi, 
This is realy good option. Previously at Vector side i used for this purpouse loki sink and labels. But i can't found a way to use nested field as lables.

Is this possible seting a nested field for `VL-Extra-Fields`?

**Comment by valyala** [comment]
FYI, v0.41.0-victorialogs has been already released - see https://docs.victoriametrics.com/victorialogs/changelog/

> Is this possible using VL-Extra-Fields to create nested field and set its value?

Yes. Just pass `source.name=foo,source.type=bar` to `VL-Extra-Fields` header or to `extra_fields` query arg as documented [here](https://docs.victoriametrics.com/victorialogs/data-ingestion/#http-parameters). Note that VictoriaLogs flattens nested JSON during data ingestion - see [these docs](https://docs.victoriametrics.com/victorialogs/keyconcepts/#data-model).

> But structured data is always parsed as nested fields. For example i can't add flat structured data to my syslog events such as:

> [source_name=source_name1 source_type=source_type1]

> victoria logs does not parse it.

This sounds like a feature, which can be easily implemented in VictoriaLogs. Could you create a separate feature request for parsing `[key1=value1 ... keyN=valueN]` syslog entries by VictoriaLogs?

**Comment by valyala** [comment]
Closing this feature request as duplicate for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7480 .

**Comment by valyala** [comment]
See also [these docs](https://docs.victoriametrics.com/victorialogs/data-ingestion/syslog/#adding-extra-fields) on how to add extra fields to Syslog logs during their ingestion into VictoriaLogs. This feature will be available in the next release.

**Comment by valyala** [comment]
The ability to add extra fields to the ingested Syslog logs is available in VictoriaLogs starting from [v0.42.0-victorialogs](https://docs.victoriametrics.com/victorialogs/changelog/) - see [these docs](https://docs.victoriametrics.com/victorialogs/data-ingestion/syslog/#adding-extra-fields).
